pkg/cli/server/server.go

```diff
@@ -364,8 +364,18 @@ func run(app *cli.Context, cfg *cmds.Server, leaderControllers server.CustomCont
 		if err != nil {
 			return errors.Wrap(err, "cannot configure default cluster-dns address")
 		}
+
 		serverConfig.ControlConfig.ClusterDNS = clusterDNS
 		serverConfig.ControlConfig.ClusterDNSs = []net.IP{serverConfig.ControlConfig.ClusterDNS}
+
+		if len(serverConfig.ControlConfig.ServiceIPRanges) > 1 {
+			// dualStack. We must pin the ipv6 address
+			extraDNS, err := utilsnet.GetIndexedIP(serverConfig.ControlConfig.ServiceIPRanges[1], 10)
+			if err != nil {
+				return errors.Wrap(err, "problems calculating extraDNS")
+			}
+			serverConfig.ControlConfig.ClusterDNSs = append(serverConfig.ControlConfig.ClusterDNSs, extraDNS)
+		}
```

Comment:
I would probably invert this logic:
1. Iterate over `serverConfig.ControlConfig.ServiceIPRanges` and store base+10 for each into `serverConfig.ControlConfig.ClusterDNSs`
2. Assign `serverConfig.ControlConfig.ClusterDNS = serverConfig.ControlConfig.ClusterDNSs[0]`

Reply:
Good idea, the code is better. And that also helps me simplify the `len(cmds.ServerConfig.ClusterDNS) != 0` case too